packages/pigeon/mock_handler_tester/test/message.dart

```diff
@@ -164,7 +235,63 @@ class Api {
         details: error['details'],
       );
     } else {
-      return SearchReply.decode(replyMap['result']!);
+      return (replyMap['result'] as SearchReply?)!;
+    }
+  }
+}
+
+class _FlutterSearchApiCodec extends StandardMessageCodec {
+  const _FlutterSearchApiCodec();
+  @override
+  void writeValue(WriteBuffer buffer, Object? value) {
+    if (value is SearchReply) {
+      buffer.putUint8(128);
+      writeValue(buffer, value.encode());
+    } else if (value is SearchRequest) {
+      buffer.putUint8(129);
+      writeValue(buffer, value.encode());
+    } else {
+      super.writeValue(buffer, value);
+    }
+  }
+
+  @override
+  Object? readValueOfType(int type, ReadBuffer buffer) {
+    switch (type) {
+      case 128:
+        return SearchReply.decode(readValue(buffer)!);
+
+      case 129:
+        return SearchRequest.decode(readValue(buffer)!);
+
+      default:
+        return super.readValueOfType(type, buffer);
+    }
+  }
+}
+
+abstract class FlutterSearchApi {
+  static const MessageCodec<Object?> codec = _FlutterSearchApiCodec();
+
+  SearchReply search(SearchRequest request);
+  static void setup(FlutterSearchApi? api) {
+    {
+      const BasicMessageChannel<Object?> channel = BasicMessageChannel<Object?>(
+          'dev.flutter.pigeon.FlutterSearchApi.search', codec);
+      if (api == null) {
+        channel.setMessageHandler(null);
+      } else {
+        channel.setMessageHandler((Object? message) async {
+          assert(message != null,
+              'Argument for dev.flutter.pigeon.FlutterSearchApi.search was null.');
+          final List<Object?> args = (message as List<Object?>?)!;
+          final SearchRequest? arg0 = args[0] as SearchRequest?;
```

Comment:
Why are these arg0, arg1, etc. rather than using the actual argument name? I know it's generated code, but it seems like it would be simple to use the name (prefixed to avoid collisions with variables you are declaring) so that someone who does read this code for some reason (say, while debugging issues that turn out to be in the generated code) would have an easier time.

Reply:
I did it on purpose to avoid any possibility for conflicting name types.  You do make a good point about being easier to debug though.

Reply:
Done.